### PR TITLE
[ENH] Agentic Iterative Evaluation: Add describe_data, score_on_holdout, and commit_estimator tools

### LINKED ISSUE #386: [ENH] Agentic forecaster workflow: missing tools for iterative candidate evaluation
## Context

I am working on an agentic forecaster prototype for the ESoC 2026 sktime track (see sktime/sktime#9721). The idea: an LLM agent runs a ReAct loop, calling sktime-mcp tools to explore forecaster candidates, score them on a holdout
window, and commit the best one. While doing this I ran into several gaps in the current tool surface.

## What I tried

I implemented an in-process version first (prototype repo: https://github.com/kpal002/sktime-agentic-forecaster) to establish the tool contract, then tried to replicate it against the sktime-mcp server. Three things are missing:

### 1. No data fingerprinting tool

The agent's first action should be to inspect the series — length, frequency, trend slope, missingness, candidate seasonal period — so it can make an informed choice before calling `list_estimators`. The existing tools expose no
equivalent. `load_data_source` + `format_time_series` prepare the data but return no summary statistics for LLM reasoning.

**Suggested tool:** `describe_data(handle)` → returns a dict with at least:
```json
{
  "length": 144,
  "frequency": "ME",
  "n_missing": 0,
  "mean": 280.3,
  "std": 78.4,
  "trend_slope_per_step": 1.2,
  "candidate_seasonal_period": 12
}
```
2. No single-candidate holdout scoring
evaluate_estimator_tool (and the proposed #369 compare_estimators_tool) run full cross-validation across many estimators at once. That is the right design for a one-shot batch call, but it does not support the iterative agent pattern where the LLM:

- Fits one candidate with specific params
- Observes the score
- Decides whether to try another variant
- Eventually commits

What is missing is a lighter-weight tool: fit a single named estimator on the in-sample portion of a data handle, score it against a held-out tail, and return the scalar metric — without running full CV or requiring the LLM to
batch all candidates upfront.

Suggested tool: score_on_holdout(handle, estimator_name, params, holdout, metric)

3. No "commit and finalize" tool
Once the agent has chosen a winner, it needs to: refit on the full training series, store the fitted model as a named handle, and surface a rationale. The current fit_predict tool trains and predicts in one call — it does not
support the "fit now, predict later at a different horizon" separation that a committed agent model needs.

A commit_estimator(handle, estimator_name, params, rationale) tool that refits on the full series and returns a persistent model handle would close this gap.

Why this matters
Without these three tools, an LLM agent using sktime-mcp cannot implement a meaningful model-selection loop. It can discover estimators and run them, but it cannot reason about data characteristics, explore candidates iteratively,
or record a justified final choice.

Issue #369 (compare_estimators_tool) partially addresses point 2 via a batch leaderboard approach. This issue proposes the complementary fine-grained tools for agents that want to explore step by step.

Happy to open a draft PR for describe_data if the design direction sounds right.

### PR DESCRIPTION
**Pull request by SHIVANSH-ux-ys**
#### Reference Issues/PRs
Fixes #386

#### What does this implement/fix? Explain your changes.

This PR implements the missing tool surface required for LLM agents to execute iterative model-selection loops (ReAct patterns) efficiently. 

It introduces three highly targeted MCP tools in a new `agentic_iterative.py` module, backed by robust `executor.py` implementations:

1. **`describe_data`:** Gives the LLM a statistical fingerprint (length, frequency, missingness, mean, std, trend proxy) of a dataset *before* it selects candidate models.
2. 2. **`score_on_holdout`:** Allows fast, single-candidate evaluation on a temporal holdout tail without the overhead of full cross-validation.
3. 3. **`commit_estimator`:** Once the agent decides on a winner, this tool refits the model on the full historical dataset, sets `fitted=True` in the handle registry, and stores an optional agent rationale.
#### Does your contribution introduce a new dependency? If yes, which one?
No new dependencies.

#### What should a reviewer concentrate their feedback on?
Reviewers can check the `Executor` class in `src/sktime_mcp/runtime/executor.py` to ensure the `temporal_train_test_split` logic aligns with `sktime` best practices.

#### Any other comments?
This completely unlocks fully autonomous agentic research loops using `sktime-mcp`!

#### PR checklist
- [x] I have read the CONTRIBUTING guide.
- [ ] - [x] I have checked that my PR does not duplicate an existing PR.
- [ ] - [x] I have checked that there is an existing issue for this PR, if applicable.
- [ ] - [x] My code follows the project's coding standards.

### CODE REVIEW

**Comment by SHIVANSH-ux-ys** [comment]
Closing this as I realized the original author of the issue is already working on the implementation for their own workflow. My apologies for the overlap!
